tests/app/views/test_suppliers.py

```diff
@@ -22,7 +22,15 @@ def setup(self):
     def teardown(self):
         self._data_api_client.stop()
 
+    def test_suppleir_details_page_requires_login(self, api_client):
+        api_client.return_value.get_supplier.return_value = self.supplier
+
+        res = self.client.get('/suppliers/1')
+
+        assert res.status_code == 302
```

Comment:
It's running with `py.test`, which rewrites assert AST to show a nice message on failure